### PR TITLE
command/cp: fixed windows filepath issue on upload

### PR DESCRIPTION
**Pull request by fbarotov**
solves issue #178

### CODE REVIEW

File: e2e/util_test.go [hunk 1]
```diff
@@ -72,8 +73,17 @@ func setup(t *testing.T, options ...option) (*s3.S3, func(...string) icmd.Cmd, f
 	for _, option := range options {
 		option(opts)
 	}
+	// testdir := fs.NewDir() tries to create a new directory which
+	// has a prefix = [test function name][operation name]
+	// e.g., prefix' = "TestCopySingleS3ObjectToLocal/cp_s3://bucket/object_file"
+	// but on windows, directories cannot contain a colon
+	// so we replace them with hyphen
+	prefix := t.Name()
+	if runtime.GOOS == "windows"{
```
Comment (sonmezonur) [inline e2e/util_test.go:82]: Seems like this code is not formatted. Please run `go fmt ./...` after each update.

File: command/cp.go [hunk 2]
```diff
@@ -395,6 +396,8 @@ func (c Copy) doUpload(ctx context.Context, srcurl *url.URL, dsturl *url.URL) er
 	size := obj.Size
 
 	if c.deleteSource {
+		// close the file before deleting
```
Comment (sonmezonur) [inline command/cp.go:399]: We can remove this comment.

File: strutil/strutil.go [hunk 3]
```diff
@@ -40,3 +42,11 @@ func JSON(v interface{}) string {
 	bytes, _ := json.Marshal(v)
 	return string(bytes)
 }
+
+// ForwardSlashify replaces all '\' with '/' on Windows OS
```
Comment (sonmezonur) [inline strutil/strutil.go:46]: Function comments should end with a period.

File: strutil/strutil.go [hunk 4]
```diff
@@ -40,3 +42,11 @@ func JSON(v interface{}) string {
...
+func ForwardSlashify(str string) string{
```
Comment (seruman) [inline strutil/strutil.go:47]: If I'm not missing anything we might be able to use [`filepath.ToSlash()` ](https://golang.org/pkg/path/filepath/#ToSlash)

> ToSlash returns the result of replacing each separator character in path with a slash ('/') character. Multiple separators are replaced by multiple slashes.

Comment (fbarotov) [reply strutil/strutil.go:47]: thanks! yes it seems filepath.ToSlash() does the job.

File: storage/url/url.go [hunk 5]
```diff
@@ -47,6 +48,7 @@ type URL struct {
 
 // New creates a new URL from given path string.
 func New(s string) (*URL, error) {
+	s = strings.ReplaceAll(s, "\\", "/")
```
Comment (sonmezonur) [inline storage/url/url.go:51]: Why we need this change?

Comment (fbarotov) [reply storage/url/url.go:51]: in windows os, to allow commands like `cp e2e\app_test.go s3://bucket-faru-1` i.e., backslashed entry

Comment (sonmezonur) [reply storage/url/url.go:51]: This function is used for creating both remote and local URLs. It is not ideal but both S3 and Unix systems may have `\\` in their files/directories. 

For Unix, `t\\est` is a valid pathname. I created an example directory:
```
tree t\\est
t\\est
|-- anotherfilewithbackslashes\\
`-- testfile

0 directories, 2 files
```


For S3, it is also valid:
```
s5cmd ls s3://bigtmp/
                                 DIR k\a/
                                 DIR ka/
2020/07/06 12:22:04                0 newtest
2020/07/06 12:21:40                0 test
```

I suspect if we do backlash replacement on the raw source,  we may affect original user input. For example, for the following command:

```
s5cmd cp test s3://bucket/prefixwith\\backslash/filewith\\backslash
# it will upload into bucket/prefixwith/backslash/filewith/backslash
```

I know it is an edge case (generally we don't use mixed file paths) and won't happen too much,  but still, we need to consider this case :) 


We should do this once we are sure that the OS is WINDOWS and the given URL is not REMOTE(S3 URL).  Maybe [here](https://github.com/peak/s5cmd/blob/master/storage/url/url.go#L53-L62) we may check OS and set URL accordingly. What do you think?

Note: We can add some simple tests for this scenario.

Comment (fbarotov) [reply storage/url/url.go:51]: Thanks a lot. Thats nice observation and suggestion. will check that too.

File: e2e/cp_test.go [hunk 6]
```diff
@@ -803,6 +807,55 @@ func TestCopyDirToS3(t *testing.T) {
...
+	defer workdir.Remove()
+	srcpath := workdir.Path()
+	dstpath := fmt.Sprintf("s3://%v/", bucket)
+
+	cmd := s5cmd("cp", workdir.Path()+"/", dstpath)
+	result := icmd.RunCmd(cmd)
+
+	result.Assert(t, icmd.Success)
+
+	fmt.Printf("src is: %s\n", srcpath)
```
Comment (sonmezonur) [inline e2e/cp_test.go:841]: We can remove logs.

File: storage/url/url.go [hunk 7]
```diff
@@ -157,13 +161,15 @@ func (u *URL) Dir() string {
...
-	if u.IsRemote() {
-		joinfn = path.Join
+	joinfn := path.Join
+
+	if runtime.GOOS == "windows" {
+		s = filepath.ToSlash(s)
 	}
 
 	clone := u.Clone()
 	clone.Path = joinfn(clone.Path, s)
```
Comment (sonmezonur) [inline storage/url/url.go:171]: If we don't need `filepath.Join`anymore, we may directly use `path.Join(clone.Path, s)` here.

File: .github/workflows/ci.yml [hunk 8]
```diff
@@ -34,7 +41,14 @@ jobs:
     - run: make test
 
   qa:
-    runs-on: ubuntu-latest
+    strategy:
```
Comment (igungor) [inline .github/workflows/ci.yml:44]: @fbarotov we should remove this matrix since `qa` doesn't contain any OS specific checks.

Comment (fbarotov) [reply .github/workflows/ci.yml:44]: got it, thanks a lot.
